### PR TITLE
rename `destroy` to `deleteMeta`

### PR DESCRIPTION
**Pull request by bekzod**
IMHO `deleteMeta` looks more explicit/clear than `destroy`

### CODE REVIEW

File: packages/ember-metal/lib/watching.js [hunk 1]
```diff
@@ -54,15 +53,3 @@ export function unwatch(obj, _keyPath, m) {
...
-  Tears down the meta on an object so that it can be garbage collected.
-  Multiple calls will have no effect.
-
-  @method destroy
-  @for Ember
-  @param {Object} obj  the object to destroy
-  @return {void}
-  @private
-*/
-export { deleteMeta as destroy };
```
Comment (bekzod) [inline packages/ember-metal/lib/watching.js]: exporting `deleteMeta` from `lib/watching.js` looked odd

File: packages/ember/lib/index.js [hunk 2]
```diff
@@ -115,7 +115,7 @@ Ember.unwatchPath = metal.unwatchPath;
 Ember.watch = metal.watch;
 Ember.isWatching = metal.isWatching;
 Ember.unwatch = metal.unwatch;
-Ember.destroy = metal.destroy;
+Ember.destroy = metal.deleteMeta;
```
Comment (bekzod) [inline packages/ember/lib/index.js]: could rename here too not sure

Comment (rwjblue) [reply packages/ember/lib/index.js]: No, we need to keep `Ember.destroy` for now.

File: packages/ember/tests/reexports_test.js [hunk 3]
```diff
@@ -96,7 +96,7 @@ QUnit.module('ember reexports');
   ['watch', 'ember-metal'],
   ['isWatching', 'ember-metal'],
   ['unwatch', 'ember-metal'],
-  ['destroy', 'ember-metal'],
+  ['deleteMeta', 'ember-metal'],
```
Comment (rwjblue) [inline packages/ember/tests/reexports_test.js]: I think this needs to be:

```js
['destroy', 'ember-metal', 'deleteMeta'],
```

The first arg is `Ember.${path}`.  The second arg is the module it comes from, and the third arg is the export from that module (if different than the first arg).

Comment (bekzod) [reply packages/ember/tests/reexports_test.js]: done

**Review by rwjblue — APPROVED** [review]

**Pull request merged**